sdk/digitaltwins/Azure.DigitalTwins.Core/src/Models/DigitalTwinMetadataJsonConverter.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Azure.DigitalTwins.Core
+{
+    /// <summary>
+    /// JSON converter to make it easier to deserialize a <see cref="BasicDigitalTwin"/>.
+    /// </summary>
+    internal class DigitalTwinMetadataJsonConverter : JsonConverter<DigitalTwinMetadata>
+    {
+        /// <inheritdoc/>
+        public override DigitalTwinMetadata Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            if (reader.TokenType == JsonTokenType.Null)
+            {
+                return null;
+            }
+
+            if (reader.TokenType != JsonTokenType.StartObject)
+            {
+                throw new JsonException($"Unexpected token type {reader.TokenType} at index {reader.TokenStartIndex} Expected JsonTokenType.StartObject.");
+            }
+
+            var metadata = new DigitalTwinMetadata();
+            reader.Read(); // Advance into our object.
+
+            // Until we reach the end of the object we began reading
+            while (reader.TokenType != JsonTokenType.EndObject)
+            {
+                string tokenName = reader.GetString();
+
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    throw new JsonException($"Unexpected token {reader.TokenType} {tokenName} found in DigitalTwinMetadata at index {reader.TokenStartIndex}, position {reader.Position}");
+                }
```

Comment:
Either this check needs to be before `reader.GetString()`, or removed. I don't think this code is reachable. When will `reader.TokenType != JsonTokenType.PropertyName` be true where the reader wouldn't have already validated and thrown before, say on `reader.Read()`?

Reply:
I'll admit I don't know JsonConverters very well. I'll send out a proposed fix and tag you on the PR for your input.